include/meltpooldg/heat/heat_transfer_operator.hpp

```diff
@@ -827,6 +854,20 @@ namespace MeltPoolDG::Heat
                      temp_vals.get_value(q_index);
             }
 
+          if (velocity && data.solidification)
+            {
+              get_material_parameter_second_derivatives_with_solidification(
+                d2_capacity_dT2, d2_density_dT2, temp_lin_vals.get_value(q_index));
+              val += (d_capacity_dT * density + d_density_dT * capacity) *
+                     (2.0 * temp_lin_vals.get_value(q_index) * temp_vals.get_gradient(q_index) +
+                      temp_vals.get_value(q_index) * temp_lin_vals.get_gradient(q_index)) *
+                     velocity_vals.get_value(q_index);
+              val += (d2_capacity_dT2 * density + 2.0 * d_capacity_dT * d_density_dT +
+                      d2_density_dT2 * capacity) *
+                     temp_lin_vals.get_value(q_index) * temp_lin_vals.get_gradient(q_index) *
+                     velocity_vals.get_value(q_index);
```

Comment:
Is here the matrix-vector product multiplication missing?
```suggestion
                     * temp_vals.get_value(q_index);
```

Reply:
You need it since you directly pass the matrix-vector product into `dst`.

Reply:
Sure, I'll add it